addons/external_intercom/functions/fnc_connect.sqf

```diff
@@ -0,0 +1,148 @@
+    ((_vehicle modelToWorld _interactionPointRelative) distance _player) > _intercomMaxRange
+    || isNil {_player getVariable "TFAR_ExternalIntercomVehicle"}
+    || (_wireless && !(_headgear isEqualTo (headgear _player)))
+}, {
+    params ["_vehicle", "_player"];
+    [_vehicle, _player] call TFAR_external_intercom_fnc_disconnect;
+}, [_vehicle, _player, _intercomMaxRange, _interactionPointRelative, _wireless, _headgear]] call CBA_fnc_waitUntilAndExecute;
+
+
+if !(_wireless) then {
+    private _handset = createSimpleObject [QPATHTOF(data\TFAR_handset.p3d), _player selectionPosition "head"];
+    _handset attachTo [_player, [-0.14,-0.02,0.02], "head", true];
+    _handset setVectorDirAndUp [[-2.5,0.8,0.25],[-1,-1,1]];
+    private _ropeID = ropeCreate [_vehicle, _interactionPointRelative vectorAdd [0,0.2,0], _handset, "plug", _intercomMaxRange]; // handset rope location[0,0.006,-0.7]
+    _vehicle setVariable ["TFAR_ExternalIntercomRopeIDs", [_ropeID, _handset], true];
+
+    _vehicle addEventHandler ["RopeBreak", {
+        params ["_vehicle", "_ropeID", "_player"];
+        if (((_vehicle getVariable ["TFAR_ExternalIntercomRopeIDs", [nil, nil]]) select 0) isEqualTo _ropeID) then {
+            [_vehicle, _player] call TFAR_external_intercom_fnc_disconnect;
+            _vehicle removeEventHandler ["RopeBreak", _thisEventHandler];
+        };
+    }];
+} else {
+
+};
+
+// Just in case the player gets into the vehicle or dies/disconnects
+_player addEventHandler ["GetInMan", {
+    params ["_player", "_role", "_vehicle"];
+    // Use 'TFAR_ExternalIntercomVehicle' variable, so that it doesn't matter
+    // what vehicle the player enters, the intercom will disconnect
+    [_player getVariable "TFAR_ExternalIntercomVehicle", _player] call TFAR_external_intercom_fnc_disconnect;
+    _player removeEventHandler ["GetInMan", _thisEventHandler];
+}];
+
+_player addMPEventHandler ["MPKilled", {
+    params ["_player"];
+    [_player getVariable "TFAR_ExternalIntercomVehicle", _player] call TFAR_external_intercom_fnc_disconnect;
+    _player removeMPEventHandler ["MPKilled", _thisEventHandler];
+}];
+
+// Just in case the vehicle is destroyed
+if (isNil {_vehicle getVariable "TFAR_ExternalIntercomEH"}) then {
+    _vehicle setVariable ["TFAR_ExternalIntercomEH",
+        _vehicle addMPEventHandler ["MPKilled", {
+            params ["_vehicle"];
+            {
+                [_vehicle, _x] call TFAR_external_intercom_fnc_disconnect;
+            } forEach ((_vehicle getVariable ["TFAR_ExternalIntercomSpeakers", [objNull, []]]) select 1);
+            _vehicle removeMPEventHandler ["MPKilled", _thisEventHandler];
+        }],
+        true
+    ];
+};
+
+_player setVariable ["TFAR_ExternalIntercomEHID",
+    ["ace_unconscious", {
+        params ["_player", "_active"];
+        if (_active) then {
+            [_player getVariable "TFAR_ExternalIntercomVehicle", _player] call TFAR_external_intercom_fnc_disconnect;
```

Comment:
Should it really disconnect if you go unconscious though?
If its wireless you still have your helmet/headset on even if you go unconscious

Reply:
Good point